### PR TITLE
Fix frequency string conversion from ts format, add test

### LINKED ISSUE #1649: _tsf_reader.convert_base doesn't handle "10 minutes" frequency correctly
## Description
For Monash datasets with the "10 minutes" frequency, the frequency converter will convert it to a frequency 10 MonthEnd (10M), instead of the expect 10 Minutes (10T) frequency.

There is already code to properly handle the "minutely" frequency, but it checks for that string explicitly, so it doesn't catch the "10 minutes" case.

## To Reproduce
One dataset which has this frequency is the 10 minutes observation Solar dataset: https://zenodo.org/record/4656144
filename: `"solar_10_minutes_dataset.zip"`
record: `"4656132"`

### PR DESCRIPTION
**Pull request by lostella**
*Issue #, if available:* Fixes #1649

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Comment by lostella** [comment]
@borchero could you check whether the problematic cases you've found are covered by the test?

**Comment by jaheba** [comment]
I wonder if we should have a more explicit lookup, ie a mapping from names to pandas frequencies. And then throw an error if we can't find the given input name.

**Comment by lostella** [comment]
> I wonder if we should have a more explicit lookup, ie a mapping from names to pandas frequencies. And then throw an error if we can't find the given input name.

Yeah makes sense, we could do that

File: src/gluonts/dataset/repository/_tsf_reader.py [hunk 1]
```diff
@@ -47,10 +49,32 @@ def frequency_converter(freq: str):
...
-    if text.lower() == "minutely":
-        return "T"
-    return text[0].upper()
+    try:
+        offset_str = BASE_FREQ_TO_PANDAS_OFFSET[text]
+    except KeyError:
+        raise GluonTSDataError(
+            f'"{text}" is not recognized as a frequency string'
+        )
+    return offset_str
```
Comment (jaheba) [inline src/gluonts/dataset/repository/_tsf_reader.py:77]:
```suggestion
    try:
        return BASE_FREQ_TO_PANDAS_OFFSET[text]
    except KeyError:
        raise GluonTSDataError(
            f'"{text}" is not recognized as a frequency string'
        )
```

File: src/gluonts/dataset/repository/_tsf_reader.py [hunk 2]
```diff
@@ -47,10 +49,31 @@ def frequency_converter(freq: str):
...
+    "minutes": "T",
+    "hourly": "H",
+    "hours": "H",
+    "daily": "D",
+    "days": "D",
+    "weekly": "W",
+    "weeks": "W",
+    "monthly": "M",
+    "months": "M",
+    "quarterly": "Q",
```
Comment (borchero) [inline src/gluonts/dataset/repository/_tsf_reader.py:64]: "quarters" for completeness?

**Review by borchero — APPROVED** [review]
LGTM

**Pull request merged**